src/bp/nlu-core/predict-pipeline.ts

```diff
@@ -199,18 +187,22 @@ async function predictContext(input: PredictStep, predictors: Predictors): Promi
   }
 }
 
-async function predictIntent(input: PredictStep, predictors: Predictors): Promise<PredictStep> {
-  if (predictors.intents.length === 0) {
-    return { ...input, intent_predictions: { per_ctx: { [DEFAULT_CTX]: [{ label: NONE_INTENT, confidence: 1 }] } } }
+async function predictIntent(input: ContextStep, predictors: Predictors): Promise<IntentStep> {
+  if (_.flatMap(predictors.intents, i => i.utterances).length <= 0) {
```

Comment:
this condition here `_.flatMap(predictors.intents, i => i.utterances).length <= 0` is taken from `engine.ts` in `makePredictors()`